### ISSUE TITLE
Add namespaces to a world

### ISSUE THREAD

**Issue by nodo** [cucumber/cucumber-ruby#1005]
## Summary

The current behaviour of `World` allows to add methods of one or more modules in a single namespace. I think it would be interesting to add different namespaces to a World, to better control which method to call and avoid name clashes.

## Proposed Solution

I would propose one of the following alternatives:

Given:
```
module MyModule
  def a_method
  end
end
```

####  1. Automatic namespace

When using `World(MyModule)` the methods of `MyModule` are accessible by using the prefix `my_module`. E.g. `my_module.my_method`. The prefix is the underscore version of the name of the module.

#### 2. Explicit namespace
`World(MyModule => :my_module)`, similar to the point above, but with a specified alias.

I have already a working prototype of this feature. The impact is not high and the new behaviour could be retro-compatible. 

The idea of my implementation is to dynamically add a method to the world which returns an instance of `Object`, which extend the module. If you think it's relevant I can share more details.

I open this issue to gather feedback and if you agree I would be happy to create a PR.

Cheers,
Andrea

**Comment by mattwynne** [comment]
Hi @nodo,

Seems like a useful idea, yes. I think it might be more idiomatic Ruby, in the explicit namespace version, to put the name first, so you can use the symbol-to-object hash syntax like this:

`World(a_module: MyModule)`

WDYT?

How about starting with a feature file in a PR, and we can discuss it more there?

**Comment by nodo** [comment]
Thanks @mattwynne, I agree with you.

I have created a feature file in an explorative PR (https://github.com/cucumber/cucumber-ruby/pull/1007). I have added a scenario to test retro-compatibility as well.

I like the idea of having two ways of adding methods two a world
1. using the old version: add helpers to a global namespace
2. using the new hash syntax: add helpers using the namespace

Looking forward to go ahead with this.
